angular-workspace/projects/ni/nimble-angular/pipes/duration.pipe.spec.ts

```diff
@@ -0,0 +1,50 @@
+import { DurationPipe } from './duration.pipe';
+
+describe('DurationPipe', () => {
+    describe('basic formatting (English)', () => {
+        let pipe: DurationPipe;
+
+        beforeEach(() => {
+            pipe = new DurationPipe('en');
+        });
+
+        const testCases = [
+            {
+                name: 'null',
+                value: null,
+                expected: ''
+            },
+            {
+                name: 'undefined',
+                value: undefined,
+                expected: ''
+            },
+            {
+                name: '1d 2h 3m 4.56s (number)',
+                value: (4.56 + 3 * 60 + 2 * 3600 + 1 * 24 * 3600) * 1000,
+                expected: '1 day, 2 hr, 3 min, 4.56 sec'
+            },
+            {
+                name: '1d 2h 3m 4.56s (as string "93784560")',
+                value: '93784560',
+                expected: '1 day, 2 hr, 3 min, 4.56 sec'
+            },
+        ];
+
+        testCases.forEach(test => {
```

Comment:
Could consider using `parameterizeNamedList` from `nimble-components`.

Reply:
I'll wait for @rajsite to weigh in on this one. The Readme beside `parameterizeNamedList` says "to be used only by nimble-component library authors", but since this is within the same repo it might be fine.

Reply:
> but since this is within the same repo it might be fine

Nah, boundary interface between projects should be what is exposed from packages. We shouldn't have hidden coupling between packages  just because they are in the same repo (nimble-angular should be able to be developed outside the repo without changes).

The options would be:

1. Move it to [`testing`](https://github.com/ni/nimble/tree/main/packages/nimble-components/src/testing) and maybe add a comment on the function saying it's internal. Create a tech debt issue to find the right spot to put it.
2. Copy and paste it to the angular lib. Create a tech debt issue to find the right spot and remove the duplicate.
3. Create a more concrete proposal for sharing it. Right now it seems like it could be as a new package as part of the style guide. See https://github.com/ni/javascript-styleguide/issues/125#issuecomment-1631682110

I think I lean to option 2 surprisingly even to me. I'm a bit concerned that option 1 would lead to nimble-components dependencies being added for the wrong reason and maybe even weird circular ones (like tokens depending on components just for the test helper). But option 1 also sounds good if it can be relatively short-term. Thoughts @jattasNI ?

Reply:
I agree that 2 is a better interim place to be than 1 and that 3 shouldn't be part of this PR, so 2 is the best of those options.

I guess there's also option 0: leave it as-is and file tech debt to create a shared library. Then we'd update this along with every other place that doesn't use parameterize today.

I think 2 puts us in a better place but if it blows up the scope I don't mind 0.

Reply:
I think my preference is not to copy/paste `parameterize` in this PR. The copied version would either be test-less, or we'd need to copy/paste the tests too. Since it's being used from the `pipes` entrypoint, I guess it would need to be exported from the `internal-utilities` entrypoint. We would also need to tweak the `tsconfig` files to allow the Jasmine types for the file it was in.

I updated #1551 to say that we should look at sharing `parameterize` with our Angular tests too (and noted that those just use `forEach` not `getSpecTypeByNamedList`).